### PR TITLE
Fix redundant move warnings by g++

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/jit/frontend/tree_views.h [hunk 1]
```diff
@@ -1262,7 +1262,11 @@ inline Expr pep604union_to_union(const Expr& expr) {
       expr.range(),
       Var::create(expr.range(), Ident::create(expr.range(), "Union")),
       List<Expr>::create(expr.range(), members));
+#if defined(__clang__)
```
Comment (Skylion007) [inline torch/csrc/jit/frontend/tree_views.h:1265]: These seems like a bug tied to a specific clang version?

Comment (cyyever) [reply torch/csrc/jit/frontend/tree_views.h:1265]: clang10 fails

**Review by ezyang — APPROVED** [review]
okkkkk

File: torch/csrc/jit/frontend/tree_views.h (hunk 1 above)
Comment (Skylion007) [reply torch/csrc/jit/frontend/tree_views.h:1265]: This is just going to cause errors in newer clang versions, isn't it?

**Pull request closed without merging**